.editorconfig

```diff
@@ -0,0 +1,25 @@
+root = true
```

Comment:
I don't think we want to check this in

Reply:
@ericstoekl apologies this was included on purpose, but isn't closely align to the PR goal 

I mentioned in the commit message
> 4. Add .editorconfig to support projects existing coding style

https://editorconfig.org/ is a widely supported by many code editors either out of the box or via a plugin

> EditorConfig helps developers define and maintain consistent coding styles between different editors and IDEs. The EditorConfig project consists of a file format for defining coding styles and a collection of text editor plugins that enable editors to read the file format and adhere to defined styles. EditorConfig files are easily readable and they work nicely with version control systems.

I've found it very helpful and thought our contributors might as well, but happy to remove if it's not something for openfaas

Reply:
I think the intentions make sense, but we use gofmt for all files, so there should technically only be one style applied.

Let's add to .gitignore for now and maybe you can demo how this works at next community meeting? If we take it in one repo, we should take it in every repo.